controllers/remotesecrets/secretbuilder.go

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remotesecrets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	api "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+	"github.com/redhat-appstudio/service-provider-integration-operator/controllers/bindings"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/remotesecretstorage"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/secretstorage"
+)
+
+type SecretBuilder struct {
```

Comment:
I'm lost in those `SecretBuilder`s. So there is `SecretBuilder` interface and 2x `SecretBuilder` implementations of that interface? All with same name? How are they different? What are they doing? Can you write short doc, maybe name them somehow so it's more clear what is it doing?

Reply:
you may have noticed that names in Go are always namespaced (unless you are doing a * import which is discouraged in most cases). I would argue that stuff like `tokenstorage.TokenStorage` is stylistically worse than `remotesecrets.SecretBuilder`.

Reply:
That said, I do agree that `SecretBuilder` is a bad name for this interface and have renamed it to an uninspired but descriptive `SecretDataGetter`.